### DISCUSSION TITLE
TeslaMate versioning and semver

### DISCUSSION THREAD

**Discussion by ramonsmits** [teslamate-org/teslamate#1197] [Q&A]
I'm updating most containers automatically via [WatchTower](https://containrrr.dev/watchtower/). I wanted to know if TeslaMate applies [Semver](https://semver.org/), meaning, breaking changes will always be introduced via majors.

Looking at https://hub.docker.com/r/teslamate/teslamate/tags there is no Major tag `1` but if you would follow SemVer user and update the `1`tag user could target `1` instead of `latest` and safely auto-update their containers via Watchtower or any alternatives or just via Docker pull.

**Comment by ramonsmits** [comment]
Unfortunate 1.27.x introduced a breaking change. This was released just when I went on holiday meaning all my trips and charging sessions were not logged as I have my containers to auto upgrade.

It would really be great if TeslaMate would follow SemVer or atleast have a docker tag that is safe to auto update.

This now happened to me twice :(

My only other option is to not auto update TeslaMate but IMHO that is not an option too.

**Comment by adriankumpf** [comment]
I'd rather adopt a calendar versioning schema, like Home Assistant. Breaking changes in the context of a whole application are a bit squishy IMO. 

The encryption of API tokens, for example, did not break anything after upgrading even if no `ENCRYPTION_KEY` env variable was set. It generated a random key and used that to encrypt the tokens. A best effort was made to store said key in presumably persistent locations like Docker volumes to cache them across application restarts. Otherwise subsequent restarts would require to log in again. 

Ultimately, whether manual intervention was necessary depended on the type of installation and setup chosen. In general, I would not recommend auto updating.
